sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/DeltaMetricStorage.java

```diff
@@ -47,9 +53,13 @@ public BoundStorageHandle bind(Attributes attributes) {
       return aggregatorHandle;
     }
 
-    // Missing entry or no longer mapped, try to add a new entry.
+    // Missing entry or no longer mapped. Try to add a new one if not exceeded cardinality limits.
     aggregatorHandle = aggregator.createHandle();
     while (true) {
+      if (activeCollectionStorage.size() >= MAX_ACCUMULATIONS) {
```

Comment:
To confirm, we've added a cap without adding any LRU behavior. Seems fine to me.

Reply:
Yes. Additionally, for synchronous instruments exporting as cumulative, metric streams are "forgotten" when they don't record during a collection.